In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import random
import scipy.stats as stats
import seaborn as sns

In [2]:
# 여기있는 데이터들이 가장 최신 rcp이다!

import os
print(os.getcwd())

/Users/we/Downloads/GCS data/user_result_final_1542090019024.csv


In [3]:
### 3개월치의 훈련데이터!  (단위 - 1주일)

chunk = {}
data1 = pd.DataFrame()
for i in range(200):
    
    chunk[i] = pd.read_csv('part-{:05d}-99e353d7-95de-430a-a54e-6c1238933495-c000.csv'.format(i))
    data1 = data1.append( chunk[i] )
    #data = data.append( chunk[i][chunk[i].rcp > 0.25] )

In [4]:
data1.shape

(28479655, 15)

In [5]:
data1.head()

mid    dcateCd    scateCd    mcateCd    lcateCd  userOrderCnt  \
0  10000002  231060205  231060200  231060000  231000000             1   
1  10000005    4102563    3100472    2100078    1100013             1   
2  10000010  233120101  233120100  233120000  233000000             1   
3  10000010  242030107  242030100  242030000  242000000             1   
4  10000020  234010303  234010300  234010000  234000000             2   

   userFirstLastOrderPeriod  orderMeanPeriod  userRecentlyOrderPeriod  \
0                         0        16.254347                       83   
1                         0        12.314815                       51   
2                         0        24.430560                       24   
3                         0        21.112414                       36   
4                        47        47.000000                       30   

                alpha                 beta    lambda      prob       rcp  \
0  0.8018117730368017   0.3499043068127323  0.048644  0.047480  0.162413   
1  0.9261682528945693  0.38539197418277665  0.093460  0.089226  0.057541   
2  0.6497197460211686  0.27510496948708457  1.990788  0.863412  0.093419   
3   0.733843713351586   0.4629269310630489  0.185879  0.169626  0.132224   
4  0.6075237709299413   0.4950685220356469  0.074026  0.071353  0.168622   

      score  
0  0.007711  
1  0.005134  
2  0.080659  
3  0.022429  
4  0.012032

# # 김태준차장님_RCP_수정_3.csv 의 결과

In [36]:
1555/6982*100   # 전체 6982개 중, 라이브딜 카테고리 1555개

22.271555428244056

In [37]:
# -rcp > 0.25 수준에서 '라이브딜'의 비율

120/409*100

29.3398533007335

In [41]:
# -rcp > 0.25 & Total user > 100 수준에서 '라이브딜'의 비율

113/232*100

48.706896551724135

In [39]:
# -rcp > 0.25 & Total user > 500 수준에서 '라이브딜'의 비율

112/169*100

66.27218934911244

In [40]:
# -rcp > 0.25 & Total user > 1000 수준에서 '라이브딜'의 비율

111/157*100

70.70063694267516

# # 11월 13일 훈련데이터에서, 1번의 구매경우도 추천해주는 경우의 category항목 조회

/Users/we/Downloads/GCS data/user_result_final_1542090019024.csv

In [6]:
data1.dcateCd.nunique()

5093

김태준차장님RCP수정_3.csv 과 왜이렇게 차이가 날까? (6982개 vs. 5093개)

- RCP구하는 기간 8개월(3월~10월), 모델 훈련기간 3개월(8월~10월) - 그만큼 딜들이 빠져나갔을 수 있다.
- 혹은, season 성 때문에, 훈련기간동안에 구매가 안이루어져서 category로서 고려되지 않았을 수도 있다!(전에 '애니메이션/교육' 에서의 case)

In [19]:
data1[(data1.dcateCd > 200000000)].dcateCd.nunique()

1258

In [20]:
data1[(data1.dcateCd > 200000000)].dcateCd.nunique() / data1.dcateCd.nunique() * 100

24.700569408992735

# # rcp 0.15 기준

In [18]:
data1[ data1.rcp > 0.15 ].dcateCd.nunique()

1050

In [17]:
data1[(data1.rcp > 0.15) & (data1.dcateCd > 200000000)].dcateCd.nunique()

349

In [21]:
349/1050*100

33.23809523809524

# # rcp 0.2 기준

In [8]:
data1[data1.rcp > 0.2].dcateCd.nunique()

548

In [22]:
data1[(data1.rcp > 0.2) & (data1.dcateCd > 200000000)].dcateCd.nunique()

198

In [23]:
198/548*100

36.13138686131387

# # rcp 0.25 기준

In [9]:
data1[data1.rcp > 0.25].dcateCd.nunique()

292

In [24]:
data1[(data1.rcp > 0.25) & (data1.dcateCd > 200000000)].dcateCd.nunique()

114

In [25]:
114/292*100

39.04109589041096

# # rcp 0.3 기준

In [10]:
data1[data1.rcp > 0.3].dcateCd.nunique()

167

In [26]:
data1[(data1.rcp > 0.3) & (data1.dcateCd > 200000000)].dcateCd.nunique()

63

In [27]:
63/167*100

37.72455089820359

# # rcp 0.35 기준

In [11]:
data1[data1.rcp > 0.35].dcateCd.nunique()

101

In [28]:
data1[(data1.rcp > 0.35) & (data1.dcateCd > 200000000)].dcateCd.nunique()

39

In [29]:
39/101*100

38.613861386138616

# # rcp 0.4 기준

In [12]:
data1[data1.rcp > 0.4].dcateCd.nunique()

59

In [30]:
data1[(data1.rcp > 0.4) & (data1.dcateCd > 200000000)].dcateCd.nunique()

19

In [31]:
19/59*100

32.20338983050847

# # rcp 0.15 기준

In [13]:
data1[data1.rcp > 0.45].dcateCd.nunique()

35

In [32]:
data1[(data1.rcp > 0.45) & (data1.dcateCd > 200000000)].dcateCd.nunique()

9

In [33]:
9/35*100

25.71428571428571

# # rcp 0.15 기준

In [14]:
data1[data1.rcp > 0.5].dcateCd.nunique()

20

In [34]:
data1[(data1.rcp > 0.5) & (data1.dcateCd > 200000000)].dcateCd.nunique()

4

In [35]:
4/20*100

20.0